dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginTest.kt

```diff
@@ -710,5 +719,84 @@ internal class DdAndroidGradlePluginTest {
         ).isFalse()
     }
 
+    @Test
+    fun `𝕄 return combined config 𝕎 resolveExtensionConfiguration() { simple variants }`(
+        @StringForgery(case = Case.LOWER) flavorA: String,
+        @StringForgery(case = Case.LOWER) flavorB: String,
+        @StringForgery(case = Case.LOWER) flavorC: String,
+        @Forgery variantConfigA: DdExtensionConfiguration,
+        @Forgery variantConfigB: DdExtensionConfiguration,
+        @Forgery variantConfigC: DdExtensionConfiguration
+    ) {
+        val flavorNames = listOf(flavorA, flavorB, flavorC)
+        variantConfigA.apply { versionName = null }
+        variantConfigB.apply { serviceName = null }
+        variantConfigC.apply { site = null }
+        mockVariant.mockFlavors(flavorNames)
+        fakeExtension.variants = mock()
+        whenever(fakeExtension.variants.findByName(flavorA)) doReturn variantConfigA
+        whenever(fakeExtension.variants.findByName(flavorB)) doReturn variantConfigB
+        whenever(fakeExtension.variants.findByName(flavorC)) doReturn variantConfigC
+
+        // When
+        val config = testedPlugin.resolveExtensionConfiguration(fakeExtension, mockVariant)
+
+        // Then
+        assertThat(config.versionName).isEqualTo(variantConfigB.versionName)
```

Comment:
just a thought: we are giving user a flexible way to define the things, but there is a possibility to shoot in the leg as well. Let's suppose that we have 2 flavors of different dimensions - `foo` and `bar`, and user wrote the following config:

```
datadog {
    site = "US"
    foo {
        serviceName = "foo"
    }
    bar {
        serviceName = "bar"
    }
}
```

In this case variant named `fooBarDebug` will get `serviceName = bar`, but maybe user would like to use `serviceName = foo`. It is a bit weird example, but I'm thinking about big enterprise apps which may have quite a lot of flavors/variants (especially if they are generated dynamically and not directly declared in the build script). Maybe it makes sense to add a check if extension field was updated more than once? wdyt?